include/hmap/dynamic-hmap.hpp

```diff
@@ -365,6 +365,10 @@ class DynamicHMap {
 		return std::make_pair(boost::make_transform_iterator<AnyCaster<V> >(iter),
 							  inserted);
 	}
+	template<typename A>
+	auto unsafe_insert_or_assign(const detail::KeyBase& kB, A&& a) {
+		return map_.insert_or_assign(kB, std::forward<A>(a));
+	}
```

Comment:
If you have a detail::KeyBase, you had V.  If you have an std::any that contains a V object, you also had V.  

If you need to insert data into an object from a dynamic key and an arbitrary expression to be evaluated to create a V, you can create that key-value pair in type-erased form **before** you forget V.  I am doing that already in the constructor invoked by make_dynamic_hmap.  It would be better to add interfaces that create and return such a pair... it would invoke "operator," of Key<V>... and that asserts a such a type-erased pair into the map... it would invoke map_.try_emplace().

Reply:
> If you need to insert data into an object from a dynamic key and an arbitrary expression to be evaluated to create a V, you can create a safe key-value pair in type-erased form before you forget V. I am doing that already in the constructor invoked by make_dynamic_hmap

This is technically true, but in the use-case that @KermMartian has for this PR, that would result in contorting the code in undesirable ways that break modularity. 

> or (b) to change the type signature of the second parameter to std:any&&.

I agree that this is the correct thing to do.

> is not appropriate because the API does not provide any way to turn that iterator into something useful... and if the caller knew the type V,

Although I don't see any immediate use case for the returned iterator, it can be fed in to a Mach7-style type switch or pattern matching construct to try and reconstruct / guess the type. This is also the intended use for the existing untemplated begin/cbegin end/cend methods.

Reply:
I was already talking about designing such a function.  It would be very easy to take a type-erased pair as an argument and check to see if it matched a particular Key<V>.  You can do a conditional cast of std::any to check types (it will return a type pointer or nullptr) and you can also check the KeyBase of the key for equality to the KeyBase in the type-erased pair.